src/minter-suite-mapping.ts

```diff
@@ -607,51 +607,39 @@ export function handleRegistrationNFTAddresses<T>(event: T): void {
   }
 }
 
-export function handleMerkleV3DelegationRegistryUpdated(
-  event: MinterMerkleV3DelegationRegistryUpdated
-): void {
-  let genericEvent: ConfigValueSetAddress;
-  genericEvent = changetype<ConfigValueSetAddress>(event);
-
-  genericEvent.parameters = [
-    // dummy value ("project ID: 0") because we're not passing in a project
-    new ethereum.EventParam("_projectId", ethereum.Value.fromI32(0)),
-    new ethereum.EventParam(
-      "_key",
-      ethereum.Value.fromBytes(Bytes.fromUTF8("delegationRegistryAddress"))
-    ),
-    new ethereum.EventParam(
-      "_value",
-      ethereum.Value.fromAddress(event.params.delegationRegistryAddress)
+export function handleDelegationRegistryUpdatedGeneric<T>(event: T): void {
+  if (
+    !(
+      event instanceof MinterMerkleV3DelegationRegistryUpdated ||
+      event instanceof MinterHolderV2DelegationRegistryUpdated
     )
-  ];
+  ) {
+    return;
+  }
 
   let minter = loadOrCreateMinter(event.address, event.block.timestamp);
 
-  handleSetValueGeneric(genericEvent, minter, null);
+  if (!minter) {
+    return;
+  }
+
+  handleSetMinterDetailsGeneric(
+    "delegationRegistryAddress",
+    event.params.delegationRegistryAddress,
+    minter
+  );
+  minter.updatedAt = event.block.timestamp;
+}
+
+export function handleMerkleV3DelegationRegistryUpdated(
+  event: MinterMerkleV3DelegationRegistryUpdated
+): void {
+  handleDelegationRegistryUpdatedGeneric(event);
 }
 export function handleHolderV2DelegationRegistryUpdated(
```

Comment:
nit: Since this is handling event from the interface, should be able to remove V2 from this function name